dashboard/modules/serve/serve_agent.py

```diff
@@ -180,6 +181,26 @@ async def put_all_applications(self, req: Request) -> Response:
                 text=repr(e),
             )
 
+        # TODO (zcin): ServeApplicationSchema still needs to have host and port
+        # fields to support single-app mode, but in multi-app mode the host and port
+        # fields at the top-level deploy config is used instead. Eventually, after
+        # migration, we should remove these fields from ServeApplicationSchema.
+        host, port = config.host, config.port
```

Comment:
Could we raise an error here if the user specifies an app-level `host` or `port`? This log is nice, but it doesn't get propagated back to the client.

Reply:
Hmm, if we want to prevent users from deploying it at all, perhaps we should just add a validator to check this in `ServeDeploySchema`.

Reply:
Yeah, that would be more general. If it doesn't impede any other behavior, then I'd prefer that.

Reply:
Agreed! I've added it [here](https://github.com/ray-project/ray/pull/33340/files#diff-e454e14e8c39af64fcec6997326bd32012fc91afe5d09d6ac1486aed6199afd1R538-R558)